app/preferences.go

```diff
@@ -64,19 +64,22 @@ func (p *preferences) load() {
 	}
 }
 
-func (p *preferences) loadFromFile(path string) error {
+func (p *preferences) loadFromFile(path string) (err error) {
 	file, err := os.Open(path) // #nosec
 	if err != nil {
 		if os.IsNotExist(err) {
-			err := os.MkdirAll(filepath.Dir(path), 0700)
-			if err != nil {
+			if err := os.MkdirAll(filepath.Dir(path), 0700); err != nil {
 				return err
 			}
 			return nil
 		}
 		return err
 	}
-	defer file.Close()
+	defer func() {
+		if r := file.Close(); r != nil {
```

Comment:
add `&& err == nil` to avoid ignoring the first error.

Reply:
```suggestion
		if r := file.Close(); r != nil && err == nil {
```